src/Renci.SshNet/Common/SshData.cs

```diff
@@ -260,8 +267,12 @@ protected string[] ReadNamesList()
         /// <summary>
         /// Reads next extension-pair data type from internal buffer.
         /// </summary>
-        /// <returns>Extensions pair dictionary.</returns>
+        /// <returns>
+        /// Extensions pair dictionary.
+        /// </returns>
+#pragma warning disable CA1859 // Use concrete types when possible for improved performance
         protected IDictionary<string, string> ReadExtensionPair()
+#pragma warning restore CA1859 // Use concrete types when possible for improved performance
```

Comment:
```suggestion
        protected Dictionary<string, string> ReadExtensionPair()
```

Reply:
That would be a breaking change. If , but I'm ok with taking the hit on this one.
What do you think?

Reply:
It's fine for me, the method is not virtual, so I can't imagine it causing any friction

Reply:
Done.